packages/sources/coinpaprika-test/src/global-utils.ts

```diff
@@ -0,0 +1,99 @@
+import { PRO_API_ENDPOINT, DEFAULT_API_ENDPOINT, customSettings } from './config'
+import { makeLogger } from '@chainlink/external-adapter-framework/util/logger'
+import { InputParameters } from '@chainlink/external-adapter-framework/validation'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+
+const logger = makeLogger('CoinPaprika Global Batched')
+
+export const inputParameters: InputParameters = {
+  market: {
+    aliases: ['to', 'quote'],
+    description: 'The symbol of the currency to convert to',
+    required: true,
+    type: 'string',
+  },
+}
+
+export interface GlobalRequestParams {
+  market: string
+}
+
+export interface GlobalResponseBody {
+  market_cap_usd: number
+  volume_24h_usd: number
+  bitcoin_dominance_percentage: number
+  cryptocurrencies_number: number
+  market_cap_ath_value: number
+  market_cap_ath_date: string
+  volume_24h_ath_value: number
+  volume_24h_ath_date: string
+  market_cap_change_24h: number
+  volume_24h_change_24h: number
+  last_updated: number
+}
+
+export type GlobalEndpointTypes = {
+  Request: {
+    Params: GlobalRequestParams
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    RequestBody: GlobalResponseBody
+    ResponseBody: GlobalResponseBody
+  }
+}
+
+export const buildGlobalRequestBody = (params: GlobalRequestParams[], apiKey?: string) => {
+  return {
+    params,
+    request: {
+      baseURL: apiKey ? PRO_API_ENDPOINT : DEFAULT_API_ENDPOINT,
+      url: '/v1/global',
+      method: 'GET',
+      headers: apiKey ? { Authorization: apiKey } : undefined,
+    },
+  }
+}
+
+export const constructEntry = (
+  res: GlobalResponseBody,
+  requestPayload: GlobalRequestParams,
+  resultPath: '_dominance_percentage' | 'market_cap_',
+) => {
+  if (!res) {
+    logger.warn(`Data not found`)
+    return
+  }
+
+  const marketMap: { [key: string]: string } = {
+    BTC: 'bitcoin',
+  }
+
+  const propertyPath =
+    resultPath === '_dominance_percentage'
+      ? `${marketMap[requestPayload.market.toUpperCase()]}${resultPath}`
+      : `market_cap_${requestPayload.market.toLowerCase()}`
```

Comment:
This is me nitpicking but just to keep both sides of the conditional consistent
```suggestion
  const propertyPath =
    resultPath === '_dominance_percentage'
      ? `${marketMap[requestPayload.market.toUpperCase()]}_dominance_percentage`
      : `market_cap_${requestPayload.market.toLowerCase()}`
```
Or you could replace `market_cap_` with `${resultPath}`. Whichever you prefer